### PR TITLE
Stop using deprecated csrf token generation

### PR DESCRIPTION
**Pull request by jordisala1991**
I am targetting this branch, because this is BC.

## Changelog

```markdown
### Fixed
- Avoid deprecation message by changing csrf token generation when possible
```

## To do

- [x] Update the tests

## Subject

To increase SF 3 compatibility, I changed the way we generate CSRF tokens with a BC layer.

Also I added some missing test classes, to start increasing code coverage, which is really low at the moment.

### CODE REVIEW

File: Controller/SecurityFOSUser1Controller.php [hunk 1]
```diff
@@ -75,10 +75,18 @@ public function loginAction()
         $lastUserNameKey = class_exists('Symfony\Component\Security\Core\Security')
             ? Security::LAST_USERNAME : SecurityContextInterface::LAST_USERNAME;
 
+        if ($this->has('security.csrf.token_manager')) { // sf >= 2.4
```
Comment (core23) [inline Controller/SecurityFOSUser1Controller.php]: Please add the `NEXT_MAJOR` comment

Comment (jordisala1991) [inline Controller/SecurityFOSUser1Controller.php]: Done, I did change some other missing NEXT_MAJOR

File: Tests/Controller/AdminSecurityControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\UserBundle\Tests\Controller;
+
+use Sonata\UserBundle\Controller\AdminSecurityController;
+use Sonata\UserBundle\Tests\Helpers\PHPUnit_Framework_TestCase;
+
+class AdminSecurityControllerTest extends PHPUnit_Framework_TestCase
```
Comment (core23) [inline Tests/Controller/AdminSecurityControllerTest.php]: 👍 for adding controller tests, but does this really test anything?

Comment (jordisala1991) [inline Tests/Controller/AdminSecurityControllerTest.php]: I added those with 2 purposes in mind:

- They do test object instantiation, so we can know if there is any syntax error or something like that.

- Having test created means that we can start testing step by step, asking contributors to test their changes.

**Comment by greg0ire** [comment]
Thanks @jordisala1991

**Pull request merged**